### PR TITLE
support double quotes in table names

### LINKED ISSUE #112: Support tablenames with double quotes "
There is no reason to not support tablenames with double quotes

The standard solution is to quote replace each one with 2 equal characters

" -> ""

This solution is backwards compatible.

### PR DESCRIPTION
**Pull request by vcalv**
Fixes #112

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -145,21 +145,17 @@ def attempt_terminate():
...
+            db.close()
 
-        db = SqliteDict(':memory:', tablename='outer space')
-        db['key'] = 'value'
-        db.commit()
-        self.assertEqual(db['key'], 'value')
-        db.close()
+        __test_irregular_tablenames('9nine')
+        __test_irregular_tablenames('outer space')
+        __test_irregular_tablenames('table with a "quoted" name')
```
Comment (piskvorky) [inline tests/test_core.py:157]: How about some non-ASCII characters?

File: sqlitedict.py [hunk 2]
```diff
@@ -161,8 +161,9 @@ def __init__(self, filename=None, tablename='unnamed', flag='c',
 
         self.filename = filename
         if '"' in tablename:
-            raise ValueError('Invalid tablename %r' % tablename)
-        self.tablename = tablename
+            self.tablename = tablename.replace('"', '""')
```
[the remark this answers is not in the document]
Comment (vcalv) [reply sqlitedict.py:164]: I was thinking of adding some logging, but without that it's OK.
In that

File: tests/test_core.py (hunk 1 above)
Comment (vcalv) [reply tests/test_core.py:158]: I don't know whether names are strings (as in unicode) or bytes. I'm assuming in 2.x it's bytes and 3.x it's unicode

I noticed there were a few bug requests mentioning problems using unicode strings as keys.

I added a test for that.

**Comment by vcalv** [comment]
BTW. `get_tablenames` returns the table names unescaped, so no need to change anything there.

File: sqlitedict.py [hunk 3]
```diff
@@ -160,15 +160,13 @@ def __init__(self, filename=None, tablename='unnamed', flag='c',
                 raise RuntimeError('Error! The directory does not exist, %s' % dirname)
 
         self.filename = filename
-        if '"' in tablename:
-            raise ValueError('Invalid tablename %r' % tablename)
-        self.tablename = tablename
+        self.tablename = tablename.replace('"', '""')
```
Comment (piskvorky) [inline sqlitedict.py:163]: Add a comment here please, + link to this ticket (for future maintainers).

File: tests/test_core.py (hunk 1 above)
Comment (piskvorky) [reply tests/test_core.py:158]: Here they are clearly strings. Do you mean inside sqlite, internally?

File: sqlitedict.py (hunk 3 above)
Comment (vcalv) [reply sqlitedict.py:167]: Linked to the relevant part of the SQL 2002 standard.

**Comment by vcalv** [comment]
FIY, sqlite identifiers are "kid of" case insensitive, but just on the ASCII characters, so:

`acute` and `Acute` are the same table, but

`ácute` and `Ácute` are different tables.

but `ácute` and `áCUTE` are the same table.

It messes with the philosophy of this project a bit if you think of table names as keys.

In particular the solution on #84 should break sometimes due to this "case insensitiveness"

File: tests/test_core.py (hunk 1 above)
Comment (vcalv) [reply tests/test_core.py:158]: I mean that in python3 calling with tablename=b'something' fails, it has to be a string, wheres in python2 calling with tablename=u'something' _might_ fail.

File: sqlitedict.py [hunk 4]
```diff
@@ -160,15 +160,19 @@ def __init__(self, filename=None, tablename='unnamed', flag='c',
                 raise RuntimeError('Error! The directory does not exist, %s' % dirname)
 
         self.filename = filename
-        if '"' in tablename:
-            raise ValueError('Invalid tablename %r' % tablename)
-        self.tablename = tablename
+
+        # standard SQL quoting of double quotes in identifiers
```
Comment (piskvorky) [inline sqlitedict.py:164]:
```suggestion
        # Use standard SQL escaping of double quote characters in identifiers, by doubling them.
```

File: sqlitedict.py [hunk 5]
```diff
@@ -160,15 +160,19 @@ def __init__(self, filename=None, tablename='unnamed', flag='c',
...
+        # ticket https://github.com/RaRe-Technologies/sqlitedict/pull/113
```
Comment (piskvorky) [inline sqlitedict.py:165]:
```suggestion
        # See https://github.com/RaRe-Technologies/sqlitedict/pull/113
```

File: sqlitedict.py [hunk 6]
```diff
@@ -160,15 +160,19 @@ def __init__(self, filename=None, tablename='unnamed', flag='c',
...
+        # for more info look at the standard @
+        # https://ronsavage.github.io/SQL/sql-2003-2.bnf.html#doublequote%20symbol
```
Comment (piskvorky) [inline sqlitedict.py:167]: These two lines are not needed (unnecessary detail). Please delete them.

**Comment by mpenkov** [comment]
Merged. Thank you for your contribution, @vcalv !

**Pull request merged**